photoeditor/src/main/java/ja/burhanrashid52/photoeditor/PhotoEditor.java

```diff
@@ -473,14 +573,16 @@ public void brushEraser() {
     }
 
     /*private void viewUndo() {
-        if (addedViews.size() > 0) {
-            parentView.removeView(addedViews.remove(addedViews.size() - 1));
+        if (viewState.addedViews.size() > 0) {
+            parentView.removeView(viewState.addedViews.remove(viewState.addedViews.size() - 1));
             if (mOnPhotoEditorListener != null)
-                mOnPhotoEditorListener.onRemoveViewListener(addedViews.size());
+                mOnPhotoEditorListener.onRemoveViewListener(viewState.addedViews.size());
         }
     }*/
 
     private void viewUndo(View removedView, ViewType viewType) {
+        final List<View> addedViews = viewState.getAddedViews();
```

Comment:
Why do we need `List<View>`. If we using this to handle undo and redo feature than better to use `Stack<View>`.

Reply:
Changed `redoViews` to a `Stack`.

Can't do this for `addedViews` because a view needs to be replaced when editing text, which I do not believe `Stack` supports.